### PR TITLE
Adding copy to FileMap functionality to languages

### PR DESCRIPTION
**Pull request by bfoley13**
Adding ability to get map[string][]byte from languages to have generated Dockerfile bytes directly

### CODE REVIEW

File: pkg/osutil/osutil.go [hunk 1]
```diff
@@ -73,19 +73,43 @@ func EnsureFile(file string) error {
 }
 
 type TemplateWriter interface {
```
Comment (imiller31) [inline pkg/osutil/osutil.go:75]: Maybe we should throw the template writer Interface and implementations in it's own go file

Comment (bfoley13) [reply pkg/osutil/osutil.go]: moved templatewriter to its own directory and created a writers directory to house the imeplemetations to not create cyclical imports

**Review by davidgamero — CHANGES_REQUESTED** [review]

File: pkg/osutil/osutil.go [hunk 2]
```diff
@@ -114,32 +138,43 @@ func CopyDir(
...
+			fileName := checkNameOverrides(f.Name(), srcPath, destPath, config)
+			if err = templateWriter.WriteFile(fmt.Sprintf("%s/%s", dest, fileName), fileString); err != nil {
 				return err
 			}
 		}
 	}
 	return nil
 }
+
+func handleTemplateReplacement(fileSys fs.FS, srcPath string, customInputs map[string]string) ([]byte, error) {
```
[the remark this answers is not in the document]
Comment (bfoley13) [reply pkg/osutil/osutil.go]: renamed

**Review by imiller31 — APPROVED** [review]
LGTM

**Review by davidgamero — APPROVED** [review]
LGTM

**Pull request merged**